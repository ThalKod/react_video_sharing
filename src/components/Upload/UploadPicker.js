import React from "react";
import ReactFilestack from 'filestack-react';


import requireLoggedIn from "../HOC/requireLoggedIn";

const UploadPicker = () => {

    const options = {
      "maxFiles": 1,
      "accept": [
        "video/mp4",
        "video/x-m4v",
        "video/*"
      ],
      "storeTo": {
        "container": "devportal-customers-assets",
        "path": "user-uploads/",
        "region": "us-east-1"
      },
      "fromSources": [
        "local_file_system",
        "url",
        "googledrive",
        "dropbox"
      ],
    };

    return(
        <div className="content-wrapper">
            <div className="container">
              <div className="row">
                <div className="col-lg-12 upload-page">
                  <div className="u-area">
                    <ReactFilestack
                      apikey={process.env.REACT_APP_API_KEY}
                      options={options}
                      onSuccess={(res) => { console.log("upload Res", res)}}
                      onError={(err) => {console.log("Upload err", err)}}
                      render={({ onPick }) => (
                        <div>
                          <i className="cv cv cvicon-cv-upload-video"/>
                          <p className="u-text1">Select Video files to upload</p>
                          <button type="submit" onClick={onPick} className="btn btn-primary u-btn">Upload Video</button>
                        </div>
                      )}
                    />
                  </div>

                  <div className="u-terms">
                    <p>By submitting your videos to circle, you acknowledge that you agree to circle's <a href="/">Terms
                      of Service</a> and <a href="/">Community Guidelines</a>.</p>
                    <p>Please be sure not to violate others' copyright or privacy rights. Learn more</p>
                  </div>
                </div>
              </div>
            </div>
        </div>
    )
};

export default requireLoggedIn(UploadPicker);


