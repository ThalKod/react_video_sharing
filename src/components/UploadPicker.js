import React from "react";

import requireLoggedIn from "./HOC/requireLoggedIn";

const UploadPicker = () => {
    return(
        <div>Hello Picker</div>
    )
};

export default requireLoggedIn(UploadPicker);


