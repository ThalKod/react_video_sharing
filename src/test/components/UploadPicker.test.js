import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import UploadPicker from "../../components/Upload/UploadPicker";

describe("UploadPicker", () => {
  it("should render correctly the UploadEdit component", () => {
    const wrapper = shallow(<UploadPicker />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
