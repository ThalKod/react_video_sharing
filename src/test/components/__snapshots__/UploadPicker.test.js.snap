// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadPicker should render correctly the UploadEdit component 1`] = `
<div
  className="content-wrapper"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-12 upload-page"
      >
        <div
          className="u-area"
        >
          <t
            apikey="ASNAiFSMAR58GpAz66oWyz"
            buttonClass=""
            buttonText="Pick file"
            cname={null}
            file={null}
            link={false}
            mode="pick"
            onError={[Function]}
            onSuccess={[Function]}
            options={
              Object {
                "accept": Array [
                  "video/mp4",
                  "video/x-m4v",
                  "video/*",
                ],
                "fromSources": Array [
                  "local_file_system",
                  "url",
                  "googledrive",
                  "dropbox",
                ],
                "maxFiles": 1,
                "storeTo": Object {
                  "container": "devportal-customers-assets",
                  "path": "user-uploads/",
                  "region": "us-east-1",
                },
              }
            }
            render={[Function]}
            security={null}
            sessionCache={false}
          />
        </div>
        <div
          className="u-terms"
        >
          <p>
            By submitting your videos to circle, you acknowledge that you agree to circle's 
            <a
              href="/"
            >
              Terms of Service
            </a>
             and 
            <a
              href="/"
            >
              Community Guidelines
            </a>
            .
          </p>
          <p>
            Please be sure not to violate others' copyright or privacy rights. Learn more
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
